core/blockchain_musicoin.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package core implements the Ethereum consensus protocol.
+package core
+
+import (
+	"errors"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// let's defeat the rat(s)
```

Comment:
https://github.com/Musicoin/go-musicoin/commit/1dc249d46cf2c04e79ff341e55d0d7a0e4208762#diff-30184e52dfee0fbd0ba21f92638b2ec1R858

Reply:
Breaking this kind of logic off into own files is something that @tzdybal and I have talked about; the intention is to simplify `merge` discrepancies later on.